web-console/4.0-designs/code-editors/code-editor.md

```diff
@@ -0,0 +1,63 @@
+# Code Editor
+
+## Edit YAML
+
+![pic 1](img/1.png)
+
+- The code area is borderless at the top and has the button bar at the bottom as a separator.
+- The Edit action can be removed from the Actions dropdown on every detail page. Users can access the tab as needed.
+- The Download and Expand links are sticky in the top right corner of the YAML editor, with a vertical separator between them. The icons belong to the left of each link.
+- Download will produce a YAML output for the user to view outside of OpenShift.
+- Expand will allow a user to view the code in fullscreen mode.
+
+![pic 2](img/2.png)
+
+- When the editor is expanded, it takes up the entire screen, covering the console entirely (including navigation and masthead).
+- The bottom button bar remains visible in the expanded state.
+- A new tab is not opened for the expanded view.
+- A collapse icon and link will be sticky in the top right corner of the expanded view for users to exit full-screen mode.
+
+![pic 2b](img/2b.png)
+
+- Some resources have additional YAML samples available for users to paste or download.
+- These samples will be accessible via a third link, "View Samples."
+- View Samples should use the help icon and should always be to the right of the Expand link.
+
+![pic 2c](img/2c.png)
+
+- Clicking the link will open a [modaless overlay](https://www.patternfly.org/pattern-library/forms-and-controls/modeless-overlay/) with the YAML samples provided. Users can close the panel using the X in the top right corner.
+
+
+#### Errors
+- Errors are shown with an icon on the corresponding line. Users can hover over the icons to see details of each error.
+
+![pic 3](img/3.png)
+
+![pic 4](img/4.png)
+
+- If errors exist, an inline notification will appear below the editor summarizing the number of errors. Users can click the Show Details link to view all error details in one place.
+- The action bar is sticky at the bottom of the page and actions include Save, Reload, and Cancel.
```

Comment:
Minor thing, but this is referred to as the button bar everywhere else in the documentation